### PR TITLE
Search .so inside jar

### PR DESCRIPTION
**Pull request by gongsu832**
- Search .so inside jar to load instead of hardcode .so name
- Remove bogus --target option on cmake when building MLIR

Signed-off-by: Gong Su <gong_su@hotmail.com>

### CODE REVIEW

**Review by caoimhinuibrian — APPROVED** [review]
LGTM

File: src/Runtime/jni/src/com/ibm/onnxmlir/OMModel.java [hunk 1]
```diff
@@ -1,58 +1,81 @@
...
+			// Copy .so to the temporary directory
+			Files.copy(jf.getInputStream(jf.getEntry(libFile)), lib,
+				   StandardCopyOption.REPLACE_EXISTING);
+
+			// z/OS USS requires "x" permission bit
+			Files.setPosixFilePermissions(lib,
+			      PosixFilePermissions.fromString("rwxr-xr-x"));
+
+			// Load the temporary .so copy
+			System.load(lib.toString());
```
Comment (seroyer) [inline src/Runtime/jni/src/com/ibm/onnxmlir/OMModel.java:56]: Is there any concept of ordering the loading of libraries?  For example if you want library foo.so loaded before bar.so?  Right now you seem to be at the mercy of the order of the files in the jar.

File: src/Runtime/jni/src/com/ibm/onnxmlir/OMModel.java [hunk 2]
```diff
@@ -1,58 +1,81 @@
...
+                                              .getLocation().toURI());
+	    Path jarDir = jar.getParent();
+
+	    // All .so files found will be copied under the ./native subdirectory
+	    // where the jar file resides.
+	    Path tmpDir = Files.createTempDirectory(jarDir, "native_");
+
+            // Open jar file to load all .so libs inside. The .so is loaded
+            // by OS so it must be extracted to the file system.
+            JarFile jf = new JarFile(jar.toFile());
```
Comment (seroyer) [inline src/Runtime/jni/src/com/ibm/onnxmlir/OMModel.java:36]: nit, there appears to be a minor indentation error here.

Comment (gongsu832) [reply src/Runtime/jni/src/com/ibm/onnxmlir/OMModel.java:36]: No problem in the file. Github is somehow displaying the diff wrong.

File: src/Runtime/jni/src/com/ibm/onnxmlir/OMModel.java (hunk 1 above)
Comment (gongsu832) [reply src/Runtime/jni/src/com/ibm/onnxmlir/OMModel.java:56]: Nothing is being resolved here. The libraries are simply being loaded to make the symbols available so it makes no difference which library is loaded first.

**Pull request merged**